### ISSUE TITLE
Explain how to actually pass inputs?

### ISSUE THREAD

**Issue by Pomax** [ad-m/github-push-action#57]
The README.md doesn't say anything about how to specify the input values. I'm sure that somewhere buried in the github action docs that's explain but _please_ put the actual steps necessary in the README.md so that people don't have to hunt for hours to find them (I know I still haven't).

**Comment by boner-cmd** [comment]
What were you trying to do? Which part is confusing?

**Comment by Pomax** [comment]
https://github.com/ad-m/github-push-action#inputs - the phrasing is "can be passed in using [...]" without explaining what "passing in" means. It looks for all the world that there isn't actually any "passing in", and the text should be updated to instead read "can be referenced using [...]"; the github token gets provided by Github, you don't pass that in yourself. The same goes for branch. So a docs update is in order.

**Comment by boner-cmd** [comment]
Your original issue stated that after hours of searching, it still wasn't clear on how to use the action **at all**. Your follow-up is that it is actually very clear on how to use the action (paste the included environmental variable) but that it isn't clear if _other_ options are available (passing in inputs without using an environmental variable, etc).

> I'm sure that somewhere buried in the github action docs that's explain but please put the actual steps necessary in the README.md

doesn't quite click with the fact that you can just paste in the suggested environmental variable without reading the github action docs at all. You can also see it from the example code provided in the doc:

> ```
> - name: Push changes
>       uses: ad-m/github-push-action@master
>       with:
>         github_token: ${{ secrets.GITHUB_TOKEN }}
>         branch: ${{ github.ref }}
> ```

Either way, this seems like a great thing for you to submit your own PR for- if you don't like the phrasing, you can resolve this yourself. However, if the fix is really as simple as changing "passed in" to "referenced using," that seems a little different from the original issue as stated, and I confess I don't really see the functional difference in the language.

Personally, if I didn't think to try using the example inputs recommended by the docs, I don't think changing the phrasing would help. But if you do, send the PR! You seem to know how to use the action now, so if you think you can add the missing link in helping others understand, you're probably in the best position to do it!

**Comment by boner-cmd** [comment]
Also worth noting that using the GitHub environmental variable for the token isn't the only valid input to reference for `github_token`- see the conversation in #23 about how that field can also refer to third party tokens. There's an existing PR to update that language it looks like (#24 )

**Comment by Pomax** [comment]
Yeah: I filed that in June. I've since figure out what the text actually meant. The original comment still stands.

**Comment by boner-cmd** [comment]
Sure, I was just suggesting that you submit your proposed changes since you think they could help other users.

**Comment by Pomax** [comment]
i contribute to 150+ repos already. Kinda done my part at this point =0

**Comment by ZPascal** [comment]
I'm closing the issue because I think the use of tokens, permissions, and secrets is already documented [here](https://github.com/ad-m/github-push-action/Readme.md#requirements-and-prerequisites). Feel free to reopen it if needed.
